containers/src/Kokkos_DynRankView.hpp

```diff
@@ -2299,9 +2299,10 @@ inline void impl_resize(const Impl::ViewCtorProp<ViewCtorArgs...>& arg_prop,
   if constexpr (alloc_prop_input::has_execution_space)
     Kokkos::Impl::DynRankViewRemap<drview_type, drview_type>(
         Impl::get_property<Impl::ExecutionSpaceTag>(prop_copy), v_resized, v);
-  else
+  else {
     Kokkos::Impl::DynRankViewRemap<drview_type, drview_type>(v_resized, v);
```

Comment:
If we're thinking about this like allocation + `deep_copy`, don't we need a global fence before this remap call, and then only a local fence after? We can do the allocation and copy stream-wise and fence them locally, but we don't know what stream may have been modifying the data in `v` before it got passed in.

Reply:
Regardless, we still need to ensure here that we're globally synchronizing before launching the remap - blocking `cudaMalloc` doesn't guarantee completion of a `parallel_for` launched on an explicit stream.

Reply:
I opened https://github.com/kokkos/kokkos/issues/6186 so we can improve this in a follow-up for all overloads.